django_heroku.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import os
+
+import dj_database_url
+from django.test.runner import DiscoverRunner
+
+MAX_CONN_AGE = 600
+
+logger = logging.getLogger(__name__)
+
+
+class HerokuDiscoverRunner(DiscoverRunner):
+    """Test Runner for Heroku CI, which provides a database for you.
+    This requires you to set the TEST database (done for you by settings().)"""
+
+    def setup_databases(self, **kwargs):
+        if not os.environ.get('CI'):
+            raise ValueError(
+                "The CI env variable must be set to enable this functionality.  WARNING:  "
+                "This test runner will wipe all tables in the 'public' schema "
+                "of the database it targets!"
+            )
+        self.keepdb = True
+        return super(HerokuDiscoverRunner, self).setup_databases(**kwargs)
+
+    def _wipe_tables(self, connection):
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                    DROP SCHEMA public CASCADE;
+                    CREATE SCHEMA public;
+                    GRANT ALL ON SCHEMA public TO postgres;
+                    GRANT ALL ON SCHEMA public TO public;
+                    COMMENT ON SCHEMA public IS 'standard public schema'; 
+                """
+            )
+        pass
+
+    def teardown_databases(self, old_config, **kwargs):
+        self.keepdb = True
+        for connection, old_name, destroy in old_config:
+            if destroy:
+                self._wipe_tables(connection)
+        super(HerokuDiscoverRunner, self).teardown_databases(old_config, **kwargs)
```

Comment:
I'm not convinced we need this at all.

Heroku CI DBs are ephemeral, so shouldn't need any cleanup.

Could you see what happens for the Heroku CI run if this is just removed entirely? (plus the corresponding `TEST_RUNNER` setting part below removed too)

Reply:
Weird error. Will discuss tmr

Reply:
Kept the class of HerokuDiscoverRunner but cut out functions _wipe_tables and teardown_databases. When trying to remove the entire functions, if self.keepdb is not kept as true the following error pops up. This has also been moved to settings.py
`-----> Running test command `python manage.py test`...
Found 1 test(s).
Creating test database for alias 'default'...
/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/postgresql/base.py:323: RuntimeWarning: Normally Django will use a connection to the 'postgres' database to avoid running initialization queries against the production database when it's not needed (for example, when running tests). Django was unable to create a connection to the 'postgres' database and will use the first PostgreSQL database instead.
  warnings.warn(
Got an error creating the test database: permission denied to create database
-----> test command `python manage.py test` failed with exit status 2`